### PR TITLE
Mask the value of the Authorization header if debug is enabled

### PR DESCRIPTION
**Pull request by rosskirkpat**
## 📝 Description

**What does this PR do and why is this change necessary?**

If `LINODE_DEBUG` is enabled, the resty debug returns the plain-text `Authorization` header value from the request. This PR ensures that the `Authorization` header value will be sanitized/masked if debug mode is enabled.

I also added a logger to the internal testutil package that is compliant with the resty.Logger interface.

## ✔️ How to Test

**What are the steps to reproduce the issue or verify the changes?**

Reproduce the issue: set `LINODE_DEBUG` when using linodego and observe a plain-text token in the debug output from resty ie. `Authorization: Bearer <LINODE_TOKEN_PLAIN_TEXT>`

Verify the changes: set `LINODE_DEBUG` when using linodego and observe a masked token in the debug output from resty (see below).
```
2024/05/07 12:58:21.997176 DEBUG RESTY 
==============================================================================
~~~ REQUEST ~~~
GET  /v4/linode/instances  HTTP/1.1
HOST   : api.linode.com
HEADERS:
	Accept: application/json
	Authorization: Bearer *******************************
	Content-Type: application/json
	User-Agent: linodego/dev https://github.com/linode/linodego
BODY   :
***** NO CONTENT *****
------------------------------------------------------------------------------
~~~ RESPONSE ~~~
STATUS       : 200
PROTO        : 
RECEIVED AT  : 2024-05-07T12:58:21.997154-04:00
TIME DURATION: 47.792µs
HEADERS      :
	Content-Type: application/json
BODY         :
{
   "id": 100,
   "region": "test-central",
   "label": "this-is-a-test-linode"
}
==============================================================================
```

**How do I run the relevant unit/integration tests?**

I added a new test `TestDebugLogSanitization` relating to these changes.

### CODE REVIEW

**Review by shanduur — APPROVED** [review]

**Review by jriddle-linode — APPROVED** [review]
Nice catch thank you.

File: client.go [hunk 1]
```diff
@@ -412,6 +414,14 @@ func (c *Client) SetHeader(name, value string) {
 	c.resty.SetHeader(name, value)
 }
 
+func (c *Client) SanitizeAuthorizationHeader() {
```
Comment (lgarber-akamai) [inline client.go:417]: Would it make sense to make this function private since it's only called from Client.SetDebug(...)?

**Review by lgarber-akamai — APPROVED** [review]
I'll be out Thursday and Friday so I'll give this my preemptive approval for once my comment has been addressed. Thanks for the contribution!

File: client.go (hunk 1 above)
Comment (rosskirkpat) [reply client.go]: I have now made this a private function

**Comment by rosskirkpat** [comment]
> I'll be out Thursday and Friday so I'll give this my preemptive approval for once my comment has been addressed. Thanks for the contribution!

@lgarber-akamai When you have a minute, would I be able ant to get your sign-off on the latest commit? I believe this PR is ready to be merged now.

**Review by lgarber-akamai — APPROVED** [review]
Looks perfect, thank you!

**Pull request merged**